src/lib/timers/ingress_drop_monitor.lua

```diff
@@ -2,48 +2,78 @@ module(...,package.seeall)
 
 -- Ingress packet drop monitor timer.
 
+local S = require("syscall")
+local counter = require("core.counter")
 local ffi = require("ffi")
+local shm = require("core.shm")
 
 -- Every 100 milliseconds.
-local interval = 1e8
+local default_interval = 1e8
 
-local with_restart = core.app.with_restart
 local now = core.app.now
 
-ingress_drop_monitor = {
-   threshold = 100000,
-   wait = 20,
-   last_flush = 0,
-   last_value = ffi.new('uint64_t[1]'),
-   current_value = ffi.new('uint64_t[1]'),
-}
+local IngressDropMonitor = {}
 
-function ingress_drop_monitor:sample ()
+function new(args)
+   local ret = {
+      threshold = args.threshold or 100000,
+      wait = args.wait or 20,
+      action = args.action or 'flush',
+      last_flush = 0,
+      last_value = ffi.new('uint64_t[1]'),
+      current_value = ffi.new('uint64_t[1]')
+   }
+   if args.counter then
+      if not args.counter:match(".counter$") then
+         args.counter = args.counter..".counter"
+      end
+      if not shm.exists("/"..S.getpid().."/"..args.counter) then
```

Comment:
I think this should be `shm.exists(args.counter)`. SHM paths are no longer resolved based on a changing `shm.path`.